plugins/slack_app/app/controllers/slack_app_controller.rb

```diff
@@ -0,0 +1,152 @@
+require 'slack'
+
+class SlackAppController < ApplicationController
+  skip_before_action :verify_authenticity_token, only: [:command, :interact]
+  skip_around_action :login_user, except: :oauth
+
+  before_action do
+    if params[:ssl_check]
+      render json: 'ok'
+    end
+
+    @payload = params[:payload] ? JSON.parse(params[:payload]) : params
+    raise "Invalid Slack token" unless @payload['token'] == ENV['SLACK_VERIFICATION_TOKEN']
+
+    @current_user_from_slack = if current_user
+      SlackIdentifier.find_by_user_id(current_user).try(:user)
+    else
+      SlackIdentifier.find_by_identifier(params[:user_id]).try(:user)
+    end
+  end
+
+  def oauth
+    @need_to_connect_app = @app_token.nil?
+    @need_to_connect_user = @current_user_from_slack.nil?
+
+    if params[:code]
+      # Got an OAuth code, let's fetch our tokens
+      resp = slack.oauth_access client_id: ENV['SLACK_CLIENT_ID'],
+                                client_secret: ENV['SLACK_CLIENT_SECRET'],
+                                code: params[:code],
+                                redirect_uri: url_for(only_path: false)
+      if @need_to_connect_app
+        @app_token = SlackIdentifier.create!(identifier: resp['access_token']).identifier
+      end
+      SlackIdentifier.create!(user_id: current_user.id, identifier: resp['user_id'])
+
+      redirect_to
+    end
+  end
+
+  def command
+    render json: command_response
```

Comment:
feels a bit strange that this method does nothing ... maybe inline command_response ?